commit/report.go

```diff
@@ -121,6 +122,11 @@ func (p *Plugin) ShouldAcceptAttestedReport(
 		return false, nil
 	}
 
+	if err := cursingValidation(ctx, p.ccipReader, decodedReport.MerkleRoots); err != nil {
+		p.lggr.Errorw("report not accepted due to cursing", "err", err)
+		return false, nil
+	}
```

Comment:
Are you sure this works? What if the transmitter can't read from the source?

Reply:
Are we only looking for curses on the destination in all cases?

Reply:
Yeah, cursing logic is as follows. If we want to curse a chain:

1. We `curse(LEGACY_CURSE_SUBJECT)` on the chain that is being cursed. This causes `Router.routeMessage` and `Router.ccipSend` to revert on that chain.
2. We `curse(bytes16(chainSelector))` on all other chains, causing `ccipSend` calls to revert from those chains to `chainSelector`.